File: unity-test/Assets/BleHidTest/Events/PairingFailedEvent.cs
using System;

namespace BleHid.Events
{
    /// <summary>
    /// Event fired when pairing with a device fails.
    /// </summary>
    public class PairingFailedEvent : DeviceEvent
    {
        /// <summary>
        /// The reason for the pairing failure
        /// </summary>
        public PairingFailureReason Reason { get; private set; }
        
        /// <summary>
        /// Creates a new pairing failed event.
        /// </summary>
        /// <param name="deviceAddress">MAC address of the device</param>
        /// <param name="reason">The reason for the pairing failure</param>
        /// <param name="eventId">ID of the event</param>
        /// <param name="javaTimestamp">Java timestamp (milliseconds since epoch)</param>
        public PairingFailedEvent(string deviceAddress, PairingFailureReason reason, Guid eventId, long javaTimestamp)
            : base(BleHidEventType.PairingFailed, deviceAddress, null, eventId, javaTimestamp)
        {
            Reason = reason;
        }
        
        /// <summary>
        /// Creates a new pairing failed event.
        /// For events created on the Unity side.
        /// </summary>
        /// <param name="deviceAddress">MAC address of the device</param>
        /// <param name="reason">The reason for the pairing failure</param>
        public PairingFailedEvent(string deviceAddress, PairingFailureReason reason)
            : base(BleHidEventType.PairingFailed, deviceAddress)
        {
            Reason = reason;
        }
        
        /// <summary>
        /// Gets a string representation of this event.
        /// </summary>
        public override string ToString()
        {
            return $"{base.ToString()}, reason={Reason}";
        }
    }
}

File: unity-plugin/src/main/java/com/example/blehid/unity/events/PairingFailedEvent.java
package com.example.blehid.unity.events;

import android.bluetooth.BluetoothDevice;

/**
 * Event fired when pairing with a device fails.
 */
public class PairingFailedEvent extends DeviceEvent {
    private final PairingFailureReason reason;
    
    /**
     * Reasons why pairing might fail
     */
    public enum PairingFailureReason {
        CANCELLED,
        TIMEOUT,
        AUTHENTICATION_FAILED,
        PROTOCOL_ERROR,
        REJECTED_BY_REMOTE,
        UNKNOWN
    }
    
    /**
     * Creates a new pairing failed event.
     * 
     * @param device The device that failed to pair
     * @param reason The reason for the pairing failure
     */
    public PairingFailedEvent(BluetoothDevice device, PairingFailureReason reason) {
        super(EventType.PAIRING_FAILED, device);
        this.reason = reason;
    }
    
    /**
     * Creates a new pairing failed event using just the address.
     * 
     * @param deviceAddress MAC address of the device that failed to pair
     * @param reason The reason for the pairing failure
     */
    public PairingFailedEvent(String deviceAddress, PairingFailureReason reason) {
        super(EventType.PAIRING_FAILED, deviceAddress);
        this.reason = reason;
    }
    
    /**
     * Gets the reason for the pairing failure.
     * 
     * @return Pairing failure reason
     */
    public PairingFailureReason getReason() {
        return reason;
    }
    
    @Override
    public String toString() {
        return String.format("%s[reason=%s]", super.toString(), reason);
    }
}
